src/hardware/vga_dac.cpp

```diff
@@ -76,12 +76,6 @@ static void vga_dac_send_color(const uint8_t palette_idx, const uint8_t color_id
 {
 	const auto rgb666 = vga.dac.rgb[color_idx];
 
-	// We might be in the middle of a mode change, so we can't use
-	// VGA_GetCurrentVideoMode() here. That's because the INT 10H mode
-	// change BIOS routine needs to set up the Palette and Color Registers
-	// which will trigger this function.
-	const auto bios_mode_number = CurMode->mode;
-
```

Comment:
I'm just moving this single line into `if (machine == MCH_VGA` as this is where the crash happens with `machine = hercules` only (because the Hercules palettes are initialised before the INT10H init stuff is called that initialised `CurMode`).

Not a great fix, but a _safe_ fix 😎 And it makes sense to restrict the `CurMode` check to the VGA case.

Reply:
Sounds reasonable.

Should we make an issue for of the underlying problem (initialisation spagetti)? Than at least that problem is documented somewhat.

Reply:
It's a known issue and I've been planning to create a refactoring issue ticket for this. Probably I'll do it after the release.